exareme2/controller/services/api/algorithm_request_validator.py

```diff
@@ -122,45 +125,87 @@ def _validate_algorithm_request_body(
     )
 
 
-def _validate_data_model(requested_data_model: str, available_datasets_per_data_model):
-    if requested_data_model not in available_datasets_per_data_model.keys():
-        raise BadUserInput(f"Data model '{requested_data_model}' does not exist.")
+def _ensure_validation_criteria(validation_datasets: List[str], validation: bool):
```

Comment:
In this context we don't know what `validation` is.

Rename to `algorithm_specification_validation_flag` ?